### ISSUE TITLE
Example build fails

### ISSUE THREAD

**Issue by Juanc1to** [cyclejs/cyclejs#964]
It looks to me like some of the dependencies for (at least) one of the examples are out of date.  Attempting to run `npm start` in "cyclejs/examples/intermediate/tsx-seconds-elapsed" fails with TypeScript errors starting with:

"node_modules/@types/node/index.d.ts(20,1): error TS1084: Invalid 'reference' directive syntax."

I updated several dependencies by deleting `dependency` and `devDependency` lines from "tsx-seconds-elapsed/package.json", and then reinstalling the associated libraries with npm, and the example now builds and runs as expected.  I'm relatively new to JavaScript and node.js development, and very new to poking around in Cycle, so I'm not sure—first—how to choose and maintain the versions of dependencies for this example, and—second—what is the best way to manage the dependency listings in "package.json".  That said, I've attached [the diff to "package.json"](https://github.com/cyclejs/cyclejs/files/5822578/packages.patch.txt), if it helps provide some information.

**Comment by jvanbruegge** [comment]
Yeah, that is a known issue at the moment. We are currently rewriting most of the internals and with that we will also redo the examples again

**Comment by Juanc1to** [comment]
That makes sense, and I'm excited to hear about the rewrite in progress!  I think I noticed it (e.g. [the Cycle.js Neo project](https://github.com/cyclejs/cyclejs/projects/5)) as I was poking through various aspects of the larger project here, and I'm quite curious about it.  What is your design and development process for this rewrite?  Is there a design document or thread somewhere that describes the new direction for the project?

With respect to the examples, would it make sense to try to maintain them in a working state until the rewrite is complete?

**Comment by jvanbruegge** [comment]
The main pull request is [here](https://github.com/cyclejs/cyclejs/pull/929). There is the design explained (including clarifications in the thread).

About the examples, if you want I would review them, but not sure how much use it is.
